.github/workflows/coverage.yml

```diff
@@ -0,0 +1,39 @@
+name: Run automated tests
+
+on: push
+
+jobs:
+  coverage:
+    name: Check coverage with coverage.py
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v4.2.2
+      - name: Install the latest version of uv
+        uses: astral-sh/setup-uv@v6.1.0
+      - name: Setup environment
+        run: uv sync --dev
+      - name: Run unit tests
+        run: uv run coverage run
+      - name: Report coverage
+        run: uv run coverage report --format markdown >> $GITHUB_STEP_SUMMARY
+      - name: Generate coverage artifacts
+        if: ${{ failure() }}
+        run: |
+          uv run coverage html --fail-under 0
+          uv run coverage xml --fail-under 0
+          uv run coverage json --fail-under 0
+          uv run coverage lcov --fail-under 0
```

Comment:
are these all the same thing in different formats? is there a reason why we capture so many formats?